### PR TITLE
Requestify `AbstractStorageDecl::hasStorage()`.

### PR DESCRIPTION
**Pull request by DougGregor**
The `hasStorage()` computation is used in many places to determine the signatures of other declarations. It currently needs to expand accessor macros, which causes a number of cyclic references. Provide a simplified request to determine `hasStorage` without expanding or resolving macros, breaking a common pattern of cycles when using macros.

Fixes rdar://109668383.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please test source compatibility

**Comment by DougGregor** [comment]
@swift-ci please clean smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: stdlib/public/Observation/Sources/Observation/Observable.swift [hunk 1]
```diff
@@ -23,19 +23,19 @@
...
+@attached(accessor, names: named(init), named(get), named(set))
 #if OBSERVATION_SUPPORTS_PEER_MACROS
 @attached(peer, names: prefixed(_))
 #endif
 public macro ObservationTracked() =
   #externalMacro(module: "ObservationMacros", type: "ObservationTrackedMacro")
 
 @available(SwiftStdlib 5.9, *)
-@attached(accessor)
+@attached(accessor, names: named(willSet))
```
Comment (Matejkob) [inline stdlib/public/Observation/Sources/Observation/Observable.swift:38]: Hi everyone,

I was reviewing this old PR and came across the `names: named(willSet)` argument. I don't get why this has been added. Does anyone recall the specific reason for adding this?

**Pull request merged**